query/context/query_context_helper.go

```diff
@@ -0,0 +1,743 @@
+//  Copyright (c) 2017-2018 Uber Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package context
+
+import (
+	memCom "github.com/uber/aresdb/memstore/common"
+	"github.com/uber/aresdb/query/common"
+	"github.com/uber/aresdb/query/expr"
+	"github.com/uber/aresdb/utils"
+	"strconv"
+	"strings"
+)
+
+// QueryContextHelper is a helper Class to group common code
+// for broker query compiler and datanode query compiler
+type QueryContextHelper struct {
```

Comment:
I feel `QueryCompilerContext` is a better name: it contains all data and methods for compiling AQL queries.

Reply:
just class name or more restructure?